### PR TITLE
Added plug and play

### PR DESCRIPTION
**Pull request by bmazerolle**
Changes
---
* Update Dockerfile to work on M1 macs
* Update nav bar with professors/create prof header group, add plug and play icon
* Added plug and play UI page
* Update all used urls to use environmental variables (will need to add a `.env` to your local build)

Testing
---
* Tested locally to update config vars -> Successful
* Deployed to heroku and tested again, [see video demo here](https://drive.google.com/file/d/1Gy-YcckUUsKkHClddKlRtHaKeqyj8u7T/view?usp=sharing)

Linked Issue
---
* https://github.com/SENG499-Company-4/Frontend/issues/73

Additional Info
---
Add this as `.env.local` to have the local build work: 
```
REACT_APP_ALGORITHM_1=company4
REACT_APP_ALGORITHM_2=company4
REACT_APP_BACKEND_URL=https://seng499company4backend.herokuapp.com/
REACT_APP_HEROKU=e2f60f10-4311-461c-8286-3b4e9eb42a1d
```

### CODE REVIEW

File: app/src/components/pages/PlugAndPlay.tsx [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+    };
+    const configVars = {
+      REACT_APP_BACKEND_URL: backendUrlInverse[backend as keyof typeof backendUrlInverse],
+      REACT_APP_ALGORITHM_1: companyInverse[algorithm1 as keyof typeof companyInverse],
+      REACT_APP_ALGORITHM_2: companyInverse[algorithm2 as keyof typeof companyInverse]
+    };
+    axios.patch('https://api.heroku.com/apps/seng499company4frontend/config-vars', configVars, config).then((res) => {
+      console.log(res);
+      cookies.remove('user');
+      window.location.href = '/';
```
Comment (sam-warren) [inline app/src/components/pages/PlugAndPlay.tsx:73]: Nit: Could use 
const navigate = useNavigate();
navigate('/'); to align with the rest of the app's nav

File: app/src/components/pages/PlugAndPlay.tsx [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+      setOldConfig(configVars);
+      setBackend(backendUrl[configVars.REACT_APP_BACKEND_URL as keyof typeof backendUrl]);
+      setAlgorithm1(company[configVars.REACT_APP_ALGORITHM_1 as keyof typeof company]);
+      setAlgorithm2(company[configVars.REACT_APP_ALGORITHM_2 as keyof typeof company]);
+      setLoading(false);
+    });
+  }, []);
+
+  if (loading) {
+    return <LoadingSpinner isLoading={true} />;
```
Comment (sam-warren) [inline app/src/components/pages/PlugAndPlay.tsx:95]: Nit: Our LoadingContext handles this now. Can wrap entire component in a LoadingWrapper instead.

File: app/src/components/shared/constants/plugandplay.constants.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+export enum company {
+  company3 = 'Company 3',
+  company4 = 'Company 4'
+}
+
+export enum companyInverse {
```
Comment (sam-warren) [inline app/src/components/shared/constants/plugandplay.constants.ts:6]: Interesting 🤔 If it works it works

**Review by sam-warren — APPROVED** [review]
Couple nit picks but we only have 3 weeks left to look a this project so it really doesn't matter.

File: app/src/components/shared/constants/plugandplay.constants.ts (hunk 3 above)
Comment (bmazerolle) [reply app/src/components/shared/constants/plugandplay.constants.ts:6]: Thats the spirit

File: app/src/components/pages/PlugAndPlay.tsx (hunk 1 above)
Comment (bmazerolle) [reply app/src/components/pages/PlugAndPlay.tsx:73]: This was actually intentional, when I clear the cookie and then use navigate() it doesnt reload the app, so itll just navigate to the home page without re-checking the now cleared cookie. As a result you'll still be able to browse the app as if you did have the cookie set. Using the `.href` method forces a refresh too

File: app/src/components/pages/PlugAndPlay.tsx (hunk 2 above)
Comment (bmazerolle) [reply app/src/components/pages/PlugAndPlay.tsx]: fixed

**Review by Ahnaf-Ahmed — APPROVED** [review]

**Review by Ahnaf-Ahmed — CHANGES_REQUESTED** [review]
Wait I think i approved too quickly. I get an axios error opon trying to reach a page 
image

**Review by Ahnaf-Ahmed — APPROVED** [review]
Sorry, the env file got deleted somehow during testing. The only question I have is whether we're going to add the register user page back in at some point?

**Pull request merged**